lib/ts/ink_defs.h

```diff
@@ -71,12 +71,27 @@
 
 // Determine the element count for an array.
 #ifdef __cplusplus
-template <typename T, unsigned N>
-static inline unsigned
-countof(const T (&)[N])
+
+#include <type_traits>
+
+template<typename T_ELEM>
+constexpr std::size_t countof()
+{
+  using T = typename std::remove_reference<T_ELEM>::type;
+  static_assert(std::is_array<T>::value, 
+                "countof() requires an array argument");
+  static_assert(std::extent<T>::value > 0,  // [0]
+                "zero- or unknown-size array");
+  return std::extent<T>::value;
+}
```

Comment:
I don't think I'm convinced that the one new use case justifies re-writing countof. Is there a major payoff I'm missing?